ports/stm32/boards/ARDUINO_PORTENTA_H7/mpconfigboard.h

```diff
@@ -0,0 +1,326 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ * The MIT License (MIT)
+ * Copyright (c) 2022 Arduino SA
+ */
+
+#define MICROPY_HW_BOARD_NAME       "PORTENTA"
+#define MICROPY_HW_MCU_NAME         "STM32H747"
+#define MICROPY_PY_SYS_PLATFORM     "Portenta"
+#define MICROPY_HW_FLASH_FS_LABEL	"portenta"
+#define MICROPY_PY_SYS_STDIO_BUFFER (0)
+#define MICROPY_STREAMS_POSIX_API   (1)
```

Comment:
I wanted to enable the btree module (I just forgot to add `MICROPY_PY_BTREE` to `mpconfigboard.mk`) but now I realized more changes are needed to enable btree in the stm32 port, so not sure what to do.

Reply:
Should I add support for btree in `stm32` port ? It requires the `posix_helpers.c` file from `esp8266` port, which can be compiled conditionally if btree module is enabled.

Reply:
I've been holding off on adding btree to stm32 because after you add something it's hard to then take it away.  btree is a good example of a library that is specialised and relatively large so you don't want to include it if you don't use it, and it is available as a native .mpy file so can be very easily installed on the filesystem when needed.

Do you have a use case for it?  Is it acceptable to use btree.mpy?

Reply:
> Do you have a use case for it? Is it acceptable to use btree.mpy?

No, but I enabled it by default for all boards because some users asked for this module specifically in the past. Are you saying there's a pure Python btree module ? If it has the same functionality I think we don't need the C module. If not, I have local changes that enable it conditionally if `MICROPY_PY_BTREE=1` is set.

Reply:
> Are you saying there's a pure Python btree module ?

There is a .mpy file that can be built and deployed using:
```
$ cd examples/natmod/btree
$ make ARCH=armv7m
$ mpremote cp btree_armv7m.mpy :btree.py
```
It is 13479 bytes in size.

Reply:
Sorry I didn't realize you're talking about a `natmod`, don't "compiled" (bytecode) modules also have a .mpy extension? Anyway, this is one option, but it requires the user to clone the repo, build mpy-cross ? build the module, copy to storage, which is a lot of work. Adding support for `btree` in `stm32` on the other hand certainly makes it a lot easier to enable builtin btree for board if it's needed.

Anyway, I removed both the POSIX_API and PY_BTREE options for now, if you want the `btree` support PR let  me know.

Reply:
> Anyway, this is one option, but it requires the user to clone the repo, build mpy-cross ? build the module, copy to storage, which is a lot of work,

It is a bit of work.  But we need to move in that direction eventually (we can't include all modules in every build) so need to find a way to make it easier for users.  Eg provide pre-build .mpy files and a web-based `mpremote` to install it.